nostrdb/src/nostrdb.c

```
@@ -5257,26 +5896,328 @@ static int ndb_write_new_blocks(struct ndb_txn *txn, struct ndb_note *note,
 
-	if (!ndb_parse_content(scratch, scratch_size, content, content_len, &blocks)) {
-		//ndb_debug("failed to parse content '%.*s'\n", content_len, content);
+	if (!giftwrap)
 		return 0;
+
+	/* looks like this is not a reprocessed giftwrap */
+	if (*ndb_note_flags(giftwrap) & NDB_NOTE_FLAG_UNWRAPPED)
+		return 1;
+
+	data = malloc(note_size);
+	memcpy(data, giftwrap, note_size);
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)